CONTRIBUTING.md

```diff
@@ -1,11 +1,310 @@
-# Contributing rulesets to HTTPS Everywhere
+# General Note
 
-If you want to create new rules to submit to us, we expect them to be in the `src/chrome/content/rules` directory. That directory also contains a useful script, `make-trivial-rule`, to create a simple rule for a specified domain. There is also a script called `utils/trivial-validate.py`, to check all the pending rules for several common errors and oversights. For example, if you wanted to make a rule for the `example.com` domain, you could run:
+There are several main areas of development on HTTPS Everywhere: the core codebase, tests, utilities, and rulesets.
+
+The core codebase is divided into two separate extensions.  One is based on the `XPCOM` API used by Mozilla Firefox (soon to be deprecated, located in the [`src`](src) top-level path), and the other is based on the `WebExtensions` API (located in [`chromium`](chromium)), first developed in Google's Chromium browser.  This consists of the code performing redirects, the UI, logging, the decentralized SSL observatory code, and ruleset loading.  This is written in JavaScript.
+
+The utilities ([`utils`](utils) top-level path) include scripts that build the extension, sanitize and perform normalization on rulesets, simplify rules, and help label GitHub issues.  Historically, these utilities have been written in Python.  Many of the newer utilities are written in JavaScript, and meant to be run in node.  Some of the wrappers for these utilities are in shell scripts.
+
+Tests are performed in headless browsers and located in the [`test`](test) top-level path.  These are written in Python, and some of the wrappers for these tests are in shell scripts.
+
+The rulesets can be found in the [`rules`](rules) top-level path include all the rules for redirecting individual sites to HTTPS.  These are written in XML.
+
+## Submitting Changes
+
+To submit changes, open a pull request from our [GitHub repository](https://github.com/efforg/https-everywhere).
+
+HTTPS Everywhere is maintained by a limited set of staff and volunteers.  Please be mindful that we may take a while before we're able to review your contributions.
+
+* * *
+
+# Contributing Code
+
+We encourage code contributions to HTTPS Everywhere.  There are a few considerations to keep in mind when contributing code.
+
+Officially supported browsers:
+
+- Firefox Stable
+- Firefox ESR
+- Firefox Dev
```

Comment:
I guess we should remove the dev channel from this documentation altogether.  We do test with Firefox Aurora, but this is to detect potential issues coming down the line.  It isn't actually officially supported.